src/test/java/oop/animals/service/EnclosureServiceTest.java

```diff
@@ -0,0 +1,167 @@
+package oop.animals.service;
+
+import oop.animals.enclosures.Enclosure;
+import oop.animals.model.mammals.Mammal;
+import oop.animals.model.mammals.predators.Predator;
+import oop.animals.model.mammals.predators.aquatic.Cachalot;
+import oop.animals.model.mammals.predators.aquatic.KillerWhale;
+import oop.animals.model.mammals.predators.ground.Coyote;
+import oop.animals.model.mammals.predators.ground.Lion;
+import oop.animals.model.mammals.predators.mixed.JungleCat;
+import oop.animals.model.mammals.predators.mixed.WhiteBear;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import java.util.ArrayList;
+import java.util.List;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+public class EnclosureServiceTest {
+    private EnclosureService<Mammal> enclosureService;
+
+    @BeforeEach
+    void setUp() throws Exception {
+        enclosureService = new EnclosureService<>();
+    }
+
+    @Test
+    public void createEnclosureTest() {
+        enclosureService.createEnclosure();
+        assertEquals(1, enclosureService.getEnclosuresSize());
+    }
+
+    @Test
+    public void createEnclosureAndAddAnimalTest() {
+        Lion lion = new Lion();
+        lion.setAge(10);
+        lion.setWeight(35);
+        lion.setWeaponType(Predator.WeaponType.TEETH);
```

Comment:
Please use constructor with parameters here after you update your model classes.